app/utils/pusher.js

```diff
@@ -38,7 +39,21 @@ TravisPusher.prototype.init = function (config, ajaxService) {
 
   return this.pusher = new Pusher(config.key, {
     encrypted: config.encrypted,
-    disableStats: true
+    disableStats: true,
+    authorizer: function (channel, options) {
+      return {
+        authorize: function (socketId, callback) {
+          let channelName = channel.name;
+          // FIXME this string was Pusher.channel_auth_endpoint, where is it from?
+          TravisPusher.ajaxService.post('/pusher/auth', {
```

Comment:
Here’s [the documentation](https://pusher.com/docs/authenticating_users#authEndpoint):

![image](https://user-images.githubusercontent.com/43280/36570156-5105ef44-17f6-11e8-8deb-625d39174d88.png)

But is there a way this could differ?

Reply:
I think that I was using `Pusher.channel_auth_endpoint`, because this is what Pusher uses by default, but I'd be fine with owning it ourselves, ie. we could move it to `config/environment.js` and define it ourselves to `/pusher/auth`

Reply:
And responding to "But is there a way this could differ?", I guess no - we have never changed it and we basically define it in the API, so in order for it to change we would need to change the API.

So I guess even hardcoding it here is fine.